### PR TITLE
Add missing error codes such as `ErrorCodeCardDeclinedRateLimitExceeded`

### PR DESCRIPTION
**Pull request by remi-stripe**
Added all the codes from [the docs](https://stripe.com/docs/error-codes. I ignored the 2 PaymentMethod ones about micro-deposits as those shouldn't be public for now.

Fixes https://github.com/stripe/stripe-go/issues/1077

r? @cjavilla-stripe 
cc @stripe/api-libraries

### CODE REVIEW

File: error.go [hunk 1]
```diff
@@ -34,18 +34,23 @@ const (
...
+	ErrorCodeBankAccountVerificationFailed          ErrorCode = "bank_account_verification_failed"
 	ErrorCodeBitcoinUpgradeRequired                 ErrorCode = "bitcoin_upgrade_required"
+	ErrorCodeCardDeclinedRateLimitExceeded          ErrorCode = "card_decline_rate_limit_exceeded"
 	ErrorCodeCardDeclined                           ErrorCode = "card_declined"
 	ErrorCodeChargeAlreadyCaptured                  ErrorCode = "charge_already_captured"
 	ErrorCodeChargeAlreadyRefunded                  ErrorCode = "charge_already_refunded"
 	ErrorCodeChargeDisputed                         ErrorCode = "charge_disputed"
 	ErrorCodeChargeExceedsSourceLimit               ErrorCode = "charge_exceeds_source_limit"
 	ErrorCodeChargeExpiredForCapture                ErrorCode = "charge_expired_for_capture"
+	ErrorCodeChargeInvalidParameter                 ErrorCode = "charge_invalid_params"
```
Comment (cjavilla-stripe) [inline error.go:50]: i think this one is `charge_invalid_parameter`

File: error.go [hunk 2]
```diff
@@ -64,6 +70,7 @@ const (
 	ErrorCodeInvalidSourceUsage                     ErrorCode = "invalid_source_usage"
 	ErrorCodeInvoiceNoCustomerLineItems             ErrorCode = "invoice_no_customer_line_items"
 	ErrorCodeInvoiceNoSubscriptionLineItems         ErrorCode = "invoice_no_subscription_line_items"
+	ErrorCodeInvoicePamentIntentRequiresAction      ErrorCode = "invoice_payment_intent_requires_action"
```
Comment (cjavilla-stripe) [inline error.go:73]: goes after `invoice_not_editable`?

File: error.go [hunk 3]
```diff
@@ -82,6 +89,7 @@ const (
 	ErrorCodeParameterMissing                       ErrorCode = "parameter_missing"
 	ErrorCodeParameterUnknown                       ErrorCode = "parameter_unknown"
 	ErrorCodeParametersExclusive                    ErrorCode = "parameters_exclusive"
+	ErrorCodePaymentIntentActionRequired            ErrorCode = "payment_intent_action_required"
 	ErrorCodePaymentIntentAuthenticationFailure     ErrorCode = "payment_intent_authentication_failure"
 	ErrorCodePaymentIntentIncompatiblePaymentMethod ErrorCode = "payment_intent_incompatible_payment_method"
 	ErrorCodePaymentIntentInvalidParameter          ErrorCode = "payment_intent_invalid_parameter"
```
Comment (cjavilla-stripe) [inline error.go:95]: Should we add `payment_method_microdeposit_verification_amounts_mismatch` and `payment_method_microdeposit_verification_attempts_exceeded` here?

Comment (remi-stripe) [reply error.go:95]: No, I mentioned both in the PR descriptions, those should not be public

Comment (cjavilla-stripe) [reply error.go:95]: ha! you mentioned these two in the comment so disregard this.

File: error.go (hunk 1 above)
Comment (remi-stripe) [reply error.go]: I wish I could rewind my life and understand what happened here since I copy-pasted each value by hand. Oh well.

**Pull request merged**